source/posix/thread.c

```diff
@@ -460,3 +460,16 @@ int aws_thread_current_at_exit(aws_thread_atexit_fn *callback, void *user_data)
     tl_wrapper->atexit = cb;
     return AWS_OP_SUCCESS;
 }
+
+struct aws_string *aws_thread_current_name(struct aws_allocator *allocator) {
+#ifdef AWS_PTHREAD_HAS_GETNAME
+    char name[16] = {0};
```

Comment:
```suggestion
    char name[256] = {0};
```
pthread_getname_np() errors with ERANGE if the buffers not big enough for the whole name

16 is the size for Linux, but I know on Mac it can get way bigger, and I believe Mac gives the main thread a pretty long name, at least it looks long in XCode.